jsonrpc/filter_manager.go

```diff
@@ -339,6 +345,70 @@ func (f *FilterManager) Exists(id string) bool {
 	return ok
 }
 
+// GetFilterLogs returns an array of logs for the specified filter.
+func (f *FilterManager) GetFilterLogs(id string) ([]*Log, error) {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+
+	result := make([]*Log, 0)
+
+	parseReceipts := func(block *types.Block, logFilter *logFilter) error {
+		receipts, err := f.store.GetReceiptsByHash(block.Header.Hash)
+		if err != nil {
+			return err
+		}
+
+		for indx, receipt := range receipts {
+			// check the logs with the filters
+			for _, log := range receipt.Logs {
+				if logFilter.query.Match(log) {
+					result = append(result, &Log{
+						Address:     log.Address,
+						Topics:      log.Topics,
+						Data:        argBytes(log.Data),
+						BlockNumber: argUint64(block.Header.Number),
+						BlockHash:   block.Header.Hash,
+						TxHash:      receipt.TxHash,
+						TxIndex:     argUint64(indx),
+					})
+				}
+			}
+		}
+
+		return nil
+	}
+
+	filter, ok := f.filters[id]
+
+	logFilters, ok := filter.(*logFilter)
+
+	if !ok {
+		return nil, ErrFilterDoesNotExists
+	}
+
+	//getAllBlocks
+	from := uint64(0)
```

Comment:
You don't need to start from zero because nothing happens when block height is zero in loop

Reply:
Agree thank you

Reply:
Completely changed the implementation so this is no longer a problem